mage/query-modules/python/gnn-link-prediction.md

```diff
@@ -0,0 +1,375 @@
+
+There are some things you should be careful about when using `link prediction`:
+- features of all nodes should be called the same(e.g saved as **'features'** property in **Memgraph**) 
+- model's performance on the validation set is obtained using **transductive** splitting mode, while **inductive** dataset split is not yet supported. You can find more information about graph splitting on slides of [Graph Machine Learning course](http://web.stanford.edu/class/cs224w/slides/08-GNN-application.pdf) offered by **Stanford**. 
+- to improve performance, `self-loop` is added to each node with the edge-type set to `self`
+- also, the user can set the flag to automatically add **reverse edges** to each existing edge and hence, convert a **directed** graph to a **bidirected** one. If the source and destination nodes of the edge are the same, **reverse edge type** will be the same as the original **edge type**. Otherwise, the prefix **rev_** will be added to the original **edge type**. See the implementation part to further see why are **self-loops** and **reverse edges** very important in ML training and how you can get into problems if your graph is already **undirected** :thinking_face: 
+- **Currently, execution on GPU-s is not supported** so setting the parameter `device_type` doesn't change anything.  
+
+Feel free to open a **[GitHub issue](https://github.com/memgraph/mage/issues)**
+or start a discussion on **[Discord](https://discord.gg/memgraph)** if you want
+to speed up development.
+
+### Usage
+
+The following procedure is expected when using **link prediction module**:
+
+- set parameters by calling `set_model_parameters()` function
+- train a model by calling `train()` function
+- inspect training results (optional) by calling `get_training_results()` function
+- predict the relationship between two vertices by calling `predict` or
+- call the `recommend` function to find the most likely relationships
+
+### Implementation details
+
+For the underlying *GNN* training we use [DGL library](https://github.com/dmlc/dgl/). 
+> Fast and memory-efficient message passing primitives for training Graph Neural Networks. Scale to giant graphs via multi-GPU acceleration and distributed training infrastructure.
+>
+> -- DGL team
+
+#### **Splitting the dataset**
+
+If the user specifies `split_ratio 1.0`, the model will train normally on a whole dataset without validating its performance on a validation set. However, if the user-defined split_ratio is a value between 0.0 and 1.0 but the graph is too small to have such a split, an exception will be thrown. 
+
+#### **Self-loops**
+
+`Self-loop edge` is added to every node to improve link_prediction performance if specified by the user. `Self-loop edges` are added only as edge_type `self`, not in any other way, and to enable this, a custom module has been added. 
```

Comment:
Why are link_prediction and edge_type written like that and not formatted or code or something?